pkgs/by-name/fo/foxglove-studio/package.nix

```diff
@@ -0,0 +1,113 @@
+{
+  alsa-lib,
+  at-spi2-atk,
+  autoPatchelfHook,
+  cairo,
+  cups,
+  dbus,
+  dpkg,
+  eudev,
+  expat,
+  fetchurl,
+  glib,
+  glibc,
+  gtk3,
+  lib,
+  libdrm,
+  libgcc,
+  libxkbcommon,
+  makeWrapper,
+  mesa,
+  nspr,
+  nss,
+  pango,
+  stdenv,
+  systemd,
+  via,
+  xorg,
+}:
+let
+  libraries = [
+    alsa-lib
+    at-spi2-atk
+    cairo
+    cups
+    dbus
+    eudev
+    expat
+    glib
+    glibc
+    gtk3
+    libdrm
+    libgcc.libgcc
+    libxkbcommon
+    mesa
+    nspr
+    nss
+    pango
+    systemd
+    via
+    xorg.libX11
+    xorg.libXcomposite
+    xorg.libXdamage
+    xorg.libXext
+    xorg.libXfixes
+    xorg.libXrandr
+    xorg.libxcb
+  ];
+in
+stdenv.mkDerivation (finalAttrs: {
+  name = "foxglove-studio";
+  version = "2.18.1";
+
+  src = fetchurl {
+    url = "https://get.foxglove.dev/desktop/v${finalAttrs.version}/foxglove-studio-latest-linux-amd64.deb";
+    sha256 = "sha256-vHloPOmrr8HR9Su5s84xohh8uCpbG4D9eEqbp+jPsDc=";
+  };
+  
+  nativeBuildInputs = [
+    autoPatchelfHook
+    dpkg
+    makeWrapper
+  ];
+
+  buildInputs = libraries;
+
+  preBuild = ''
+    addAutoPatchelfSearchPath $out/opt/Foxglove Studio/
+  '';
+
+  unpackPhase = ''
+    dpkg-deb -x $src .
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out/bin
+    cp -R usr/share opt $out/
+
+    # Entrypoint
+    mkdir -p $out/bin
+    makeWrapper "$out/opt/Foxglove Studio/foxglove-studio" $out/bin/foxglove-studio \
+      --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath libraries}"
```

Comment:
I don't think we need to have both a wrapper and autoPatchelfing

Reply:
Mhm I got this from this very nice [blog](https://lantian.pub/en/article/modify-computer/nixos-packaging.lantian/#examples-closed-source-software--binary-distributed-ones) the `Medium: DingTalk (Auto Patch Binaries, Finding Dependencies)` example. 
I thought because I'm not sure if foxglove dynamically loads binaries I included it since it won't hurt. 
But if you are sure it doesn't i can remove it.

Reply:
I removed it now! I checked and Im confident that foxglove does`t load dynamic libs